src/clr-angular/data/datagrid/datagrid-row.ts

```diff
@@ -168,8 +172,10 @@ export class ClrDatagridRow<T = any> implements AfterContentInit, AfterViewInit
 
   ngAfterContentInit() {
     this.dgCells.changes.subscribe(() => {
-      this.dgCells.forEach(cell => {
-        this._scrollableCells.insert(cell._view);
+      this.dgCells.forEach((cell, index) => {
+        if (this._scrollableCells.indexOf(cell._view) === -1) {
+          this._scrollableCells.insert(cell._view, this.columnReorderService.orderAt(index));
+        }
```

Comment:
Missing `this.subscriptions.push()` also recommend moving logic into a descriptive method
Something like this maybe?

```typescript
this.subscriptions.push(this.updateScrollableCellsOnCellChanges());
```

Reply:
Thanks. will fix it.